torch/ao/quantization/pt2e/numeric_debugger.py

```diff
@@ -11,3 +20,188 @@ def generate_numeric_debug_handle(graph_module: GraphModule) -> None:
+        if NUMERIC_DEBUG_HANDLE_KEY not in n.meta:
+            continue
+        numeric_debug_handle = n.meta[NUMERIC_DEBUG_HANDLE_KEY]
+        _insert_logger(model, n, numeric_debug_handle)
+
+    model.recompile()
+    return model
+
+@dataclass(frozen=True)
+class QuantizationComparisonResult:
+    actual: torch.Tensor
+    ref: torch.Tensor
+
+    @property
+    def mse_loss(self) -> torch.Tensor:
+        return F.mse_loss(
+            self.actual.to(dtype=torch.float32), self.ref.to(dtype=torch.float32)
+        )
+
+    @property
+    def sqnr(self) -> torch.Tensor:
+        return compute_sqnr(
+            self.actual.to(dtype=torch.float32), self.ref.to(dtype=torch.float32)
+        )
+
+    def details(self) -> str:
+        return f"QuantizationComparisonResult(mse_loss={self.mse_loss}, sqnr={self.sqnr}, actual={self.actual}, ref={self.ref})"
+
+    def __repr__(self) -> str:
+        # Don't include the tensors themselves as they are quite large to print
+        # out.
+        return (
+            f"QuantizationComparisonResult(mse_loss={self.mse_loss}, sqnr={self.sqnr})"
+        )
+
+
+@dataclass(frozen=True)
+class NodeAccuracySummary:
+    handle: str
+    actual_node_name: str
+    actual_module_stack: str
+    ref_node_name: str
+    ref_module_stack: str
+    results: Sequence[QuantizationComparisonResult]
+
+    def details(self) -> str:
+        """Returns a very detailed string for the results including all tensors.
+        If you don't want to see this, just use str() or repr() on this object instead
+        to get a more concise summary."""
+        s = f"{repr(self)}\ndetails:\n"
+        for q in self.results:
+            s += q.details() + "\n"
+        return s
+
+
+def _module_stack_to_str(module_stack: object) -> str:
+    if not isinstance(module_stack, dict):
+        return str(module_stack)
+    module_values_list = list(module_stack.values())
+    if len(module_values_list) > 0:
+        owning_module = module_values_list[-1][0]
+        return str(owning_module)
+    else:
+        return str(module_stack)
+
+
+def extract_results_from_loggers(
+    ref_model: GraphModule,
+    actual_model: GraphModule,
+) -> Dict[str, NodeAccuracySummary]:
+    """For each model, extract the comparison of tensors for each debug handle.
+    The first model is the reference, the second is after applying a transform.
+
+    Returns:
+        A dict is keyed by the node name from the reference (first) model given,
+        and the values are a summary of all the tensors that passed through each
+    logger and statistics about how different they are from each other."""
+    # Results maps debug handle to a tensor list for each model being compared.
+    ref_handles: Dict[str, Tuple[str, object, List[torch.Tensor]]] = {}
+    actual_handles: Dict[str, Tuple[str, object, List[torch.Tensor]]] = {}
+    for _name, module in ref_model.named_children():
+        if isinstance(module, OutputLogger):
```

Comment:
I recommend checking if `len(module.stats) > 0` here, to avoid comparing empty loggers. Maybe a warning can be emitted in that case, the user didn't run `forward`